src/client/sandbox/node/window.ts

```diff
@@ -654,28 +656,33 @@ export default class WindowSandbox extends SandboxBase {
                             : new DOMException('Only secure origins are allowed.', 'SecurityError'));
                     }
 
-                    args[0] = getProxyUrl(url, { resourceType: stringifyResourceType({ isScript: true }) });
+                    args[0] = getProxyUrl(url, { resourceType: stringifyResourceType({ isServiceWorker: true }) });
                 }
 
-                if (args[1] && typeof args[1].scope === 'string') {
-                    args[1].scope = getProxyUrl(args[1].scope, {
-                        resourceType: stringifyResourceType({ isScript: true })
-                    });
-                }
+                args[1] = { scope: '/' };
+
+                return nativeMethods.registerServiceWorker.apply(window.navigator.serviceWorker, args)
+                    .then(reg => {
+                        const parsedProxyUrl = parseProxyUrl(args[0]);
+                        const serviceWorker  = reg.installing || reg.waiting || reg.active;
```

Comment:
Should we postpone sending the `set-settings` message until a service worker became active?

Reply:
`ServiceWorkerRegistration` has no events for this.